### PR TITLE
feat(components): add label and validation hint to RadioButtonGroup

### PR DESCRIPTION
**Pull request by Inkyov**
## Purpose

Add label and validation error to the RadioButtonGroup component

## Approach and changes

- Add `label`
 - Add `InlineMessage` to show the `ValidationHint`

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: src/components/RadioButtonGroup/RadioButtonGroup.stories.tsx [hunk 1]
```diff
@@ -20,26 +20,52 @@ import { RadioButtonGroup, RadioButtonGroupProps } from './RadioButtonGroup';
 export default {
   title: 'Forms/RadioButton/RadioButtonGroup',
   component: RadioButtonGroup,
-  argTypes: {
-    name: { control: 'text' },
-    value: { control: 'text' },
-  },
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.stories.tsx:26]: Why did you remove this information?

File: src/components/RadioButtonGroup/RadioButtonGroup.stories.tsx [hunk 2]
```diff
@@ -20,26 +20,52 @@ import { RadioButtonGroup, RadioButtonGroupProps } from './RadioButtonGroup';
...
-    { children: 'Banana', value: 'banana' },
-    { children: 'Mango', value: 'mango' },
-  ],
+export const Base = () => (
+  <RadioButtonGroupWithState name="radio-button-group" />
+);
+
+export const Invalid = (args: RadioButtonGroupProps) => (
+  <RadioButtonGroupWithState {...args} />
+);
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.stories.tsx:66]: I know it's tempting to extract the shared logic into a base component here. However, inside story files, it's better not to do so, since the story source code is displayed underneath each story in the docs. If you extract the main logic away, it becomes hidden from the docs.

So, please copy and paste the code from the base component into each individual story :)

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 3]
```diff
@@ -40,34 +47,61 @@ export interface RadioButtonGroupProps extends HTMLProps<HTMLDivElement> {
    * The ref to the HTML Dom element
    */
   ref?: Ref<HTMLDivElement>;
+  /**
+   * Warning or error message, displayed in a tooltip.
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.tsx:51]: I know this is probably copied from another input component, but it's no longer accurate 🙈
```suggestion
   * Warning or error message, displayed below the radio buttons.
```

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 4]
```diff
@@ -14,9 +14,16 @@
...
+import { css } from '@emotion/core';
 
+import styled, { StyleProps } from '../../styles/styled';
 import { uniqueId } from '../../util/id';
 import { RadioButton, RadioButtonProps } from '../RadioButton/RadioButton';
+import Label from '../Label';
+import {
+  InlineMessage,
+  InlineMessageProps,
+} from '../InlineMessage/InlineMessage';
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.tsx:26]: The InlineMessage component is meant for _form_ level errors. _Field_ level errors should use the [ValidationHint](https://github.com/sumup-oss/circuit-ui/tree/canary/src/components/ValidationHint) component.

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 5]
```diff
@@ -40,34 +47,61 @@ export interface RadioButtonGroupProps extends HTMLProps<HTMLDivElement> {
...
-              {...{ ...rest, value, name, onChange }}
-              checked={value === activeValue}
-            >
-              {children}
-            </RadioButton>
-          </div>
-        ))}
-    </div>
+    <Label htmlFor={name}>
+      {label && <span>{label}</span>}
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.tsx:84]: The Label component / `label` HTML element is meant for single inputs only. For field groups, you should use the `fieldset` and `legend` HTML elements (see this [implementation example](https://dequeuniversity.com/library/aria/radio-and-radio-group)).

You can still use the Label component for the styles though: `<Label as="legend">`.

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 6]
```diff
@@ -40,34 +47,61 @@ export interface RadioButtonGroupProps extends HTMLProps<HTMLDivElement> {
...
+                {children}
+              </RadioButton>
+            </div>
+          ))}
+      </div>
+      {invalid && (
+        <StyledInlineMessage variant="danger">
+          {validationHint}
+        </StyledInlineMessage>
+      )}
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.tsx:103]: Once you've replaced the InlineMessage component with the ValidationHint component, you could also add support for the `hasWarning` and `showValid` props that are supported by other input components.

**Review by connor-baer — CHANGES_REQUESTED** [review]
Thank you for this valuable contribution @Inkyov! 

There are a few issues with the markup. Let me know if you have questions about my comments.

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx (hunk 4 above)
Comment (Inkyov) [reply src/components/RadioButtonGroup/RadioButtonGroup.tsx:26]: Yeah I placed the ValidationHint component but then Bobi (our designer) said that it's better to place the InlineMessage. 🤷

Comment (connor-baer) [reply src/components/RadioButtonGroup/RadioButtonGroup.tsx:26]: What was his reason for it? 

This simply goes against the usage guidelines for the InlineMessage. It's for form level errors inside a card only. I know that it's being used differently/incorrectly in the Dashboard and Shop currently, we're trying to correct those cases as well. 

Please use the ValidationHint.

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 7]
```diff
@@ -49,25 +66,41 @@ const RadioButtonGroupComponent = (
...
-              {...{ ...rest, value, name, onChange }}
-              checked={value === activeValue}
-            >
-              {children}
-            </RadioButton>
-          </div>
-        ))}
-    </div>
+    <fieldset name={name}>
+      {label && <legend>{label}</legend>}
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.tsx:81]: The styling of the legend should match other input labels, so it should use the `kilo` text size (you could use the `textKilo` style helper) and have a `bit` sized bottom margin.

File: src/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 8]
```diff
@@ -49,25 +66,41 @@ const RadioButtonGroupComponent = (
...
+
+      <div role="group" aria-label={label} ref={ref} {...props}>
```
Comment (connor-baer) [inline src/components/RadioButtonGroup/RadioButtonGroup.tsx:83]: Now that the radio buttons are wrapped inside the fieldset, this div has become obsolete. Please remove it and move the `ref` and `{...props}` props to the fieldset element. The `role` and `aria-label` props are no longer needed.

**Review by connor-baer — CHANGES_REQUESTED** [review]

**Review by connor-baer — APPROVED** [review]
Perfect 🥇 Thank you so much for polishing this!
